### DISCUSSION TITLE
Can I change the path feishin grabs its playlists from?

### DISCUSSION THREAD

**Discussion by FatMax1492** [jeffvli/feishin#791] [Q&A] [answered]
The playlists I have in jellyfin are not showing up in Feishin, probably because I create m3u files in Musicbee and allow Jellyfin access to the folder they're saved in. 

Does Feishin read m3u files and can I thusly change Feishin's playlist path to conform to my file structure?

**Comment by jeffvli** [comment, marked as answer]
Playlist imports are handled by your server itself (in this case Jellyfin), Feishin doesn't read anything from your filesystem except for the settings files.

**Reply by FatMax1492** [reply]
Ah, that's too bad.

Thanks anyways

**Reply by mihawk90** [reply]
> Playlist imports are handled by your server itself (in this case Jellyfin)

I also mentioned it on Discord, but for the sake of public discourse:

I do believe that in this case this is actually an issue with Feishin.

M3U (and M3U8) playlists are imported and handled by Jellyfin just fine, but they don't show in Feishin.

When I first ran into this I believed this to be an issue with either the Jellyfin API not returning those playlists or it being an issue with the new public/private playlists in Jellyfin 10.9.
But whatever it is, evidently it *is* possible to pull these playlists, because [Finamp](https://github.com/jmshrv/finamp) is indeed showing them, while Feishin is not.

I'll attach my screenshots from Discord, first a view of the JF WebUI (top) and Feishin (bottom), and second a screenshot from Finamp:
![image](https://github.com/user-attachments/assets/8f7ed7e8-b460-45e4-8157-cfa924aa289d)
![image](https://github.com/user-attachments/assets/e5c3c52f-26f0-4392-aed1-e8e9ac04d4e4)

---

Essentially, OP is asking the wrong question:
They are asking to look for the M3U elsewhere when the real question should be: "Why are my imported M3U not showing up?"

**Reply by FatMax1492** [reply]
You're probably right, but the answer to the question still remains valid. 

Imported m3u files are handled as a seperate thing by Jellyfin. Only 'native' playlists get a corresponding xml file in the settings folder. It's only there where Feishin looks for playlists, and thus imported playlists won't be shown. 

I do however think it would be nice to have Feishin support imported playlists.

**Reply by jeffvli** [reply]
I just [pushed a change](https://github.com/jeffvli/feishin/commit/3a63ee4b95d76e495b96e701d7ebd52941258e5a) that should fix the issue with m3u playlists not showing up in Feishin.

**Reply by mihawk90** [reply]
> Only 'native' playlists get a corresponding xml file in the settings folder. It's only there where Feishin looks for playlists, and thus imported playlists won't be shown.

No, that's inaccurate. Feishin doesn't have access to the filesystem. The whole point of Jellyfin is to be an abstraction layer between physical files on the filesystem and mediaplayers.

Mediaplayers such as Feishin just request a list of things (e.g. playlists) and Jellyfin returns results. The mediaplayer never sees the physical files underneath.

Yes m3u files are probably handled differently by Jellyfin internally, but that doesn't mean mediaplayers can't list them (see Finamp and the Feishin change mentioned above), nor that they get access to the filesystem.

Arguably this is likely a bug in Jellyfin. IMHO imported m3u files should inherently be categorised as MediaType Audio when they are imported from a music library path, and therefore returned when requested. But that doesn't seem to be the case as of right now.

**Reply by FatMax1492** [reply]
oh, well it's working now anyways. 

Thanks!
